### PR TITLE
Lvnod 127 adjusting jest for testing react component and redux

### PR DESCRIPTION
**Pull request by avoytovich**
## JIRA

* [Main JIRA ticket](https://ssu-jira.softserveinc.com/browse/LVNOD-XXX)
* [JIRA subtask](https://ssu-jira.softserveinc.com/browse/LVNOD-XXX)

## Code reviewers

- [x] @yarvanya
- [x] @AlbaMontis
- [x] @shyyka

### Second Level Review

- [ ] @lhalam
- [ ] @4asof4ik

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo
## CHECK LIST
- [x]  СI passed
- [ ]  All reviewers agreed to merge the PR

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -15,12 +15,39 @@
...
+    "babel-preset-react": "^6.24.1",
+    "expect": "^1.20.2",
+    "identity-obj-proxy": "^3.0.0",
+    "jest": "^20.0.4",
+    "mock-local-storage": "^1.0.2",
+    "nock": "^9.0.13",
+    "react-test-renderer": "^15.6.1",
     "redux-logger": "^3.0.6",
+    "redux-mock-store": "^1.2.3",
+    "redux-thunk": "^2.2.0",
```
Comment (AlbaMontis) [inline package.json]: Look at "dependencies".

File: package.json [hunk 2]
```diff
@@ -15,12 +15,39 @@
...
+    "setupTestFrameworkScriptFile": "mock-local-storage"
+  },
   "bugs": {
     "url": "https://github.com/Lv-237-NodeJS/da-mi/issues"
   },
   "homepage": "https://github.com/Lv-237-NodeJS/da-mi#readme",
   "devDependencies": {
+    "babel-jest": "^20.0.3",
+    "babel-preset-es2015": "^6.24.1",
+    "babel-preset-react": "^6.24.1",
```
Comment (AlbaMontis) [inline package.json]: Look at "dependencies".

File: package.json [hunk 3]
```diff
@@ -15,12 +15,39 @@
...
+    ],
+    "setupTestFrameworkScriptFile": "mock-local-storage"
+  },
   "bugs": {
     "url": "https://github.com/Lv-237-NodeJS/da-mi/issues"
   },
   "homepage": "https://github.com/Lv-237-NodeJS/da-mi#readme",
   "devDependencies": {
+    "babel-jest": "^20.0.3",
+    "babel-preset-es2015": "^6.24.1",
```
Comment (AlbaMontis) [inline package.json]: Look at "dependencies".

File: test/Login_async_action_creators.test.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+    router.browserHistory = { push: () => {} };
+
+    const expectedActions = [
+      loginActions.loginUserRequest(),
+      loginActions.loginUserSuccess('do something', 777)
+    ];
+
+    const store = mockStore({ loggedUser: [] });
+
+    nock(API.HOST + API.PORT)
```
Comment (AlbaMontis) [inline test/Login_async_action_creators.test.js]: Use API.URL.

File: test/Login_async_action_creators.test.js [hunk 5]
```diff
@@ -0,0 +1,46 @@
+import * as loginActions from './../app/src/redux/login';
+import { API } from './../app/src/helper/constants';
+import * as router from 'react-router';
+import configureMockStore from 'redux-mock-store';
+import thunk from 'redux-thunk';
+import nock from 'nock';
+import expect from 'expect';
+
+const middlewares = [thunk];
```
Comment (AlbaMontis) [inline test/Login_async_action_creators.test.js:9]: Do you need this if you have only one middleware?

File: package.json (hunk 3 above)
Comment (avoytovich) [reply package.json]: i took it from jest guide for adjusting work. If you know another way, could you tell me?

File: package.json (hunk 2 above)
Comment (avoytovich) [reply package.json]: if you want that i understand something you will have to explain more simply

File: package.json (hunk 1 above)
Comment (avoytovich) [reply package.json]: what do you mean?

File: test/Login_async_action_creators.test.js (hunk 5 above)
Comment (avoytovich) [reply test/Login_async_action_creators.test.js:9]: i know only this way...

File: package.json (hunk 3 above)
Comment (AlbaMontis) [reply package.json]: We already have this package in "dependencies".

File: package.json (hunk 2 above)
Comment (AlbaMontis) [reply package.json]: We already have this package in "dependencies".

File: package.json (hunk 1 above)
Comment (AlbaMontis) [reply package.json]: We already have this package in "dependencies".

File: test/Login_async_action_creators.test.js [hunk 6]
```diff
@@ -0,0 +1,46 @@
...
+        const response = Object.assign(JSON.parse(res), {token: 'do something', user_id: 777});
+        const token = response.token;
+        const userId = response.user_id;
+        return store.dispatch(loginActions.loginUserSuccess(token, userId));
+      });
+
+    store.dispatch(loginActions.loginUser('ivan.yarymovych@gmail.com', 'P!assword!1'));
+
+    expect(store.getActions()).toEqual(expectedActions);
+
```
Comment (4asof4ik) [inline test/Login_async_action_creators.test.js]: drop

File: test/Login_component.test.js [hunk 7]
```diff
@@ -0,0 +1,47 @@
+import React from 'react';
+import renderer from 'react-test-renderer';
+import { Login } from './../app/src/components/Login/Login';
+import { loginUser } from './../app/src/redux/login';
+
+describe('React component <Login />', () => {
+
```
Comment (4asof4ik) [inline test/Login_component.test.js]: drop

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: test/Login_async_action_creators.test.js (hunk 6 above)
Comment (avoytovich) [reply test/Login_async_action_creators.test.js]: what must i drop? Sorry, but i don't understand what do you mean...

File: test/Login_component.test.js (hunk 7 above)
Comment (avoytovich) [reply test/Login_component.test.js]: what do you mean? Could you explain me in some simply form...thank's...

File: test/Login_async_action_creators.test.js (hunk 6 above)
Comment (avoytovich) [reply test/Login_async_action_creators.test.js]: Done

File: package.json (hunk 3 above)
Comment (avoytovich) [reply package.json]: Done

File: package.json (hunk 2 above)
Comment (avoytovich) [reply package.json]: Done

File: package.json (hunk 1 above)
Comment (avoytovich) [reply package.json]: Done

File: test/Login_component.test.js (hunk 7 above)
Comment (avoytovich) [reply test/Login_component.test.js]: Done

**Review by yarvanya — DISMISSED** [review]

File: test/__snapshots__/Login_component.test.js.snap [hunk 8]
```diff
@@ -0,0 +1,189 @@
...
+    className="login-form form-horizontal"
+    onSubmit={[Function]}
+    state={
+      Object {
+        "email": "",
+        "password": "",
+      }
+    }
+  >
+    <div
```
Comment (shyyka) [inline test/__snapshots__/Login_component.test.js.snap:15]: is the necessary to brake elements with one attribute?

**Review by shyyka — DISMISSED** [review]

File: app/src/components/Login/Login.js [hunk 9]
```diff
@@ -30,17 +30,17 @@ class Login extends React.Component {
...
     e.preventDefault();
     this.props.actions.loginUser(this.state.email, this.state.password);
-  }
+  };
 
   render () {
     return (
       <div>
-        <Form className='login-form' horizontal onSubmit={this.handleSubmit}>
+        <Form className='login-form' horizontal onSubmit={this.handleSubmit} state = {this.state}>
```
Comment (AlbaMontis) [inline app/src/components/Login/Login.js]: Useless space around "=".

**Review by AlbaMontis — DISMISSED** [review]

File: app/src/components/Login/Login.js (hunk 9 above)
Comment (avoytovich) [reply app/src/components/Login/Login.js]: Done

**Review by yarvanya — DISMISSED** [review]

File: test/__snapshots__/Login_component.test.js.snap (hunk 8 above)
Comment (avoytovich) [reply test/__snapshots__/Login_component.test.js.snap:15]: maybe) this is a snapshots...its created auto...

**Review by yarvanya — DISMISSED** [review]

**Review by AlbaMontis — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

File: test/Login_component.test.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+      actions: {
+        loginUser
+      }
+    };
+    const component = renderer.create(
+      <Login {...user} />
+    );
+    let tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
+    const form = tree.children.filter(child => child.type === 'form')[0];
```
Comment (4asof4ik) [inline test/Login_component.test.js]: add empty line

File: test/Login_component.test.js [hunk 11]
```diff
@@ -0,0 +1,40 @@
...
+    password.props.onChange({
+      target: {
+        value: 'P!assword!1'
+      }
+    });
+    form.props.onSubmit({
+      preventDefault: () => {},
+    });
+    tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
```
Comment (4asof4ik) [inline test/Login_component.test.js]: and here

File: package.json [hunk 12]
```diff
@@ -15,12 +15,36 @@
...
   "author": "",
   "license": "ISC",
+  "jest": {
+    "moduleFileExtensions": [
+      "es6",
+      "js"
+    ],
+    "verbose": true,
+    "moduleNameMapper": {
+      "\\.(jpg|jpeg|png|gif|eot|otf|webp|svg|ttf|woff|woff2|mp4|webm|wav|mp3|m4a|aac|oga)$": "<rootDir>/__mocks__/fileMock.js",
```
Comment (4asof4ik) [inline package.json]: you have all this type of file?

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: test/Login_component.test.js (hunk 10 above)
Comment (avoytovich) [reply test/Login_component.test.js]: Done

File: test/Login_component.test.js (hunk 11 above)
Comment (avoytovich) [reply test/Login_component.test.js]: Done

File: package.json (hunk 12 above)
Comment (avoytovich) [reply package.json]: Done...reduced some

**Review by yarvanya — DISMISSED** [review]

**Review by AlbaMontis — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by AlbaMontis — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by yarvanya — DISMISSED** [review]

**Review by shyyka — DISMISSED** [review]

**Review by AlbaMontis — DISMISSED** [review]

File: test/Login_reducers.test.js [hunk 13]
```diff
@@ -0,0 +1,68 @@
+import loginReduser from './../app/src/redux/login';
+import * as loginActions from './../app/src/redux/login';
+import * as router from 'react-router';
+
+describe('loginReduser execution', () => {
```
Comment (4asof4ik) [inline test/Login_reducers.test.js:5]: describe(this is not test)(after first describe no need empty line)
    describe (this is not test)
        it (this is test)
          (empty line after each test)
       it
  (empty line after each describe)
    describe
       it

       it


fix in this format

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

File: test/Login_reducers.test.js (hunk 13 above)
Comment (avoytovich) [reply test/Login_reducers.test.js:5]: done

File: test/Login_async_action_creators.test.js [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+import expect from 'expect';
+
+const middlewares = [thunk];
+const mockStore = configureMockStore(middlewares);
+
+describe('async action creators', () => {
+  afterEach(() => {
+    nock.cleanAll();
+  });
+  it('creates LOGIN_USER_SUCCESS when user has been logged in', () => {
```
Comment (4asof4ik) [inline test/Login_async_action_creators.test.js]: додай пустий рядок перед it

File: test/Login_async_action_creators.test.js [hunk 15]
```diff
@@ -0,0 +1,37 @@
...
+        password: 'P!assword!1',
+      })
+      .reply(201, (uri, res) => {
+        const response = Object.assign(JSON.parse(res), {token: 'do something', user_id: 777});
+        const token = response.token;
+        const userId = response.user_id;
+        return store.dispatch(loginActions.loginUserSuccess(token, userId));
+      });
+    store.dispatch(loginActions.loginUser('ivan.yarymovych@gmail.com', 'P!assword!1'));
+    expect(store.getActions()).toEqual(expectedActions);
```
Comment (4asof4ik) [inline test/Login_async_action_creators.test.js]: додай пустий рядок перед  expect

File: test/Login_component.test.js [hunk 16]
```diff
@@ -0,0 +1,40 @@
...
+    const user = {
+      actions: {
+        loginUser
+      }
+    };
+    const component = renderer.create(
+      <Login {...user} />
+    );
+    let tree = component.toJSON();
+    expect(tree).toMatchSnapshot();
```
Comment (4asof4ik) [inline test/Login_component.test.js]: додай пустий рядок перед expect

File: test/Login_component.test.js (hunk 11 above)
Comment (4asof4ik) [reply test/Login_component.test.js]: додай пустий рядок перед expect

File: test/Login_reducers.test.js [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+  const state = {};
+  describe('LOGIN_USER_FAILURE', () => {
+    it('should return the correct state', () => {
+      const action = loginActions.loginUserFailure('message');
+      expect(loginReduser(state, action))
```
Comment (4asof4ik) [inline test/Login_reducers.test.js]: додай пустий рядок перед expect

File: test/Login_reducers.test.js [hunk 18]
```diff
@@ -0,0 +1,57 @@
...
+          illegalInput: true,
+          message: action.message
+        });
+    });
+  });
+
+  describe('CHECK_TOKEN', () => {
+    it('should return the correct state', () => {
+      const action = loginActions.checkToken();
+      expect(loginReduser(state, action)).toEqual({
```
Comment (4asof4ik) [inline test/Login_reducers.test.js]: додай пустий рядок перед expect

File: test/Login_reducers.test.js [hunk 19]
```diff
@@ -0,0 +1,57 @@
...
+        ...state,
+        isAuth: action.isAuth
+      });
+    });
+  });
+
+  describe('LOGOUT_USER', () => {
+    it('should return the correct state', () => {
+      router.browserHistory = { push: () => {} };
+      expect(loginReduser(state, loginActions.logout())).toEqual({
```
Comment (4asof4ik) [inline test/Login_reducers.test.js]: додай пустий рядок перед expect

File: test/Login_reducers.test.js [hunk 20]
```diff
@@ -0,0 +1,57 @@
...
+        isAuth: false
+      });
+    });
+  });
+
+  describe('LOGIN_USER_SUCCESS', () => {
+    it('should return the correct state', () => {
+      router.browserHistory = { push: () => {} };
+      const action = loginActions.loginUserSuccess('token', 777);
+      expect(loginReduser(state, action)).toEqual({
```
Comment (4asof4ik) [inline test/Login_reducers.test.js]: додай пустий рядок перед expect

**Review by 4asof4ik — CHANGES_REQUESTED** [review]

**Review by shyyka — APPROVED** [review]

**Review by 4asof4ik — APPROVED** [review]

**Pull request merged**